waku/peer.go

```diff
@@ -224,7 +258,7 @@ func (p *Peer) broadcast() error {
 	envelopes := p.host.Envelopes()
 	bundle := make([]*Envelope, 0, len(envelopes))
 	for _, envelope := range envelopes {
-		if !p.marked(envelope) && envelope.PoW() >= p.powRequirement && p.bloomMatch(envelope) {
+		if !p.marked(envelope) && envelope.PoW() >= p.powRequirement && p.topicOrBloomMatch(envelope) {
```

Comment:
Wait, do I understand correctly that it will apply both topic and bloom filter matching? It should only apply one or the other. Topics prioritised if they are set. Similar as I think is done in `topicInterestOrBloomMatch`.

Reply:
I though it was doing that, as we don't set `fullNode` to true if `topicInterest` is set, but just looking at the function `BloomFilterMatch` is matches if bloom filter is nil, regardless on whether `fullNode` is set. I'll add a test to ensure the behavior is correct.